client/src/components/pages/HomePage/HomePage.tsx

```diff
@@ -5,85 +5,105 @@ import ErrorBoundary from 'react-error-boundary';
+const filterSchema = Configuration[METADATA]['filters'];
+const SearchResult = Configuration[SEARCH_RESULT];
 
 const getSearchFilters = (searchResults: SearchArticle[] | null): SearchFilters => {
-  if (searchResults === null || searchResults.length === 0) {
-    return defaultFilter;
-  }
-
-  let min = Number.MAX_VALUE;
-  let max = -1;
-  let authors: Set<string> = new Set([]);
-  let journals: Set<string> = new Set([]);
-  let sources: Set<string> = new Set([]);
-
-  searchResults.forEach((article) => {
-    if (article.publish_time) {
-      const year = Number(article.publish_time.substr(0, 4));
-      min = Math.min(year, min);
-      max = Math.max(year, max);
-    }
-
-    if (article.authors) {
-      article.authors.forEach((a) => authors.add(a));
-    }
-
-    if (article.source) {
-      article.source.forEach((s) => sources.add(s));
+  // Iterate through JSON fields
+  const fields = Object.keys(filterSchema);
+  let filterValues: SearchFilters = {};
+  fields.forEach((filter) => {
+    // Handle year filter values
+    if (filterSchema[filter].type === 'year_slider') {
+      if (searchResults === null || searchResults.length === 0) {
+        filterValues[filter] = [0, 0]; // Dummy range
+      } else {
+        let min = Number.MAX_VALUE;
+        let max = -1;
+        searchResults.forEach((article) => {
+          if (article[filter]) {
+            const year = Number(article[filter].substr(0, 4));
+            min = Math.min(year, min);
+            max = Math.max(year, max);
+          }
+        });
+        //  Show monthly granularity if only 1 year range
+        filterValues[filter] = min === max ? [min * 100 + 1, min * 100 + 12] : [min, max];
+      }
+    } else if (filterSchema[filter].type === 'selection') {
+      filterValues[filter] = new Set([]);
+      if (searchResults !== null && searchResults.length > 0) {
+        searchResults.forEach((article) => {
+          const val = article[filter];
+          if (!val) {
+            return;
+          } else if (typeof val === 'string' || val instanceof String) {
+            filterValues[filter].add(val);
+          } else if (Array.isArray(val)) {
+            article[filter].forEach((a: String) => filterValues[filter].add(a));
+          }
+        });
+        filterValues[filter] = Array.from(filterValues[filter].values()).filter(
+          (a: any) => a.length > 0,
+        );
+      }
     }
+  });
+  return filterValues;
+};
 
-    if (article.journal) {
-      journals.add(article.journal);
+const filterArticle = (selectedFilters: any, article: SearchArticle): Boolean => {
+  let includeArticle = true;
+  const fields = Object.keys(selectedFilters);
+  fields.forEach((field) => {
+    const val = article[field];
+    if (filterSchema[field].type === 'year_slider') {
+      includeArticle =
+        includeArticle &&
+        (!val ||
+          (val.length >= 7 &&
+            Number(val.substr(0, 7).replace('-', '')) >= selectedFilters[field][0] &&
```

Comment:
Is this month and year based filtering?

Reply:
Yeah only if month is available, we do this for covidex if the year range is small